retry.go

```diff
@@ -52,40 +61,47 @@ import (
 	"time"
 )
 
+// Function signature of retry if function
+type RetryIfFunc func(error) bool
+
 // Function signature of retryable function
 type Retryable func() error
 
 // Function signature of OnRetry function
 // n = count of tries
 type OnRetry func(n uint, err error)
 
-// Retry - simple retry
-func Retry(retryableFunction Retryable) error {
-	return RetryWithOpts(retryableFunction, NewRetryOpts())
-}
+func Do(retryableFunction Retryable, opts ...Option) error {
+	var n uint
 
-// RetryWithOpts - customizable retry via RetryOpts
-func RetryWithOpts(retryableFunction Retryable, opts RetryOpts) error {
-	return RetryCustom(retryableFunction, func(n uint, err error) {}, opts)
-}
+	//default
+	config := &config{
+		tries:           10,
+		delay:           1e5,
+		onRetryFunction: func(n uint, err error) {},
+		retryIfFunction: func(err error) bool { return true },
+	}
 
-// RetryCustom - the most customizable retry
-// is possible set OnRetry function callback
-// which are called each retry
-func RetryCustom(retryableFunction Retryable, onRetryFunction OnRetry, opts RetryOpts) error {
-	var n uint
+	//apply opts
+	for _, opt := range opts {
+		opt(config)
+	}
 
-	errorLog := make(Error, opts.tries)
+	errorLog := make(Error, config.tries)
 
-	for n < opts.tries {
+	for n < config.tries {
 		err := retryableFunction()
 
 		if err != nil {
-			onRetryFunction(n, err)
+			config.onRetryFunction(n, err)
 			errorLog[n] = err
 
-			delayTime := opts.delay * (1 << (n - 1))
-			time.Sleep((time.Duration)(delayTime) * opts.units)
+			if !config.retryIfFunction(err) {
```

Comment:
Pokud nespecifikuji retryIfFuction, tak se mi to opakovani neprovede? To bych nevynucoval. Pokud je specifikovana, tak ji provolam a vyhodnotim, jinak bych udelal ten pocet opakovani, co je nastaveny.

Reply:
default je ze se to opakuje...